### PR TITLE
[#207] Rename app name in strings.xml

### PR DESCRIPTION
**Pull request by Wadeewee**
#207

## What happened 👀

Rename app name within all strings.xml files inside the project folder according to app-name.

## Proof Of Work 📹

https://user-images.githubusercontent.com/28002315/169774187-3b4967a9-0cd6-4386-b6cd-26f0abcd85ad.mov

### CODE REVIEW

File: scripts/new_project.kts [hunk 1]
```diff
@@ -149,6 +151,20 @@ object NewProject {
             }
     }
 
+    private fun renameAppName() {
+        showMessage("=> 🔎 Renaming app name...")
+        File(projectPath)
+            ?.walk()
+            .filter { it.isFile && it.name == "strings.xml" }
```
Comment (Tuubz) [inline scripts/new_project.kts:158]: Could it be that `it.isFile` is redundant? 🙏

File: scripts/new_project.kts [hunk 2]
```diff
@@ -8,6 +8,7 @@ object NewProject {
     private const val KEY_PACKAGE_NAME = "package-name"
     private const val TEMPLATE_FOLDER_NAME = "CoroutineTemplate"
     private const val TEMPLATE_PACKAGE_NAME = "co.nimblehq.coroutine"
+    private const val TEMPLATE_APP_NAME = "Coroutine Template"
```
Comment (Tuubz) [inline scripts/new_project.kts:11]: We could put it before `TEMPLATE_FOLDER_NAME` to keep alphabetical order 🚀

**Review by Tuubz — COMMENTED** [review]
Rest LGTM 👍

**Review by manh-t — COMMENTED** [review]
just have the same comments with @Tuubz , rest LGTM

File: scripts/new_project.kts [hunk 3]
```diff
@@ -9,23 +9,67 @@ object NewProject {
...
     private const val TEMPLATE_PACKAGE_NAME = "co.nimblehq.coroutine"
 
+    private const val APP_PATTERN = "^[A-Z\$][a-zA-Z0-9\$]*\$"
+    private const val PACKAGE_PATTERN = "^[a-z]+(\\.[a-z][a-z0-9]*)*\$"
+
     private val modules = listOf("app", "data", "domain")
 
     private var appName = ""
-    private val appNameWithoutSpace: String
-        get() = appName.replace(" ", "")
```
Comment (kaungkhantsoe) [inline scripts/new_project.kts:16]: I think we need this `appNameWithoutSpace` don't we?

Comment (Wadeewee) [reply scripts/new_project.kts:16]: You're right @kaungkhantsoe After rebase this case will be solved 🚀

**Comment by Tuubz** [comment]
@Wadeewee Please let us know once you have rebased, so we can review again 🚀

**Review by Tuubz — APPROVED** [review]
🔥 🔥 🔥

**Review by manh-t — APPROVED** [review]
🚒 🚒 🧯

**Review by kaungkhantsoe — APPROVED** [review]

**Pull request merged**